### ISSUE TITLE
.dumbjump: Outside path adds unnecessary context-free search results

### ISSUE THREAD

**Issue by cweiske** [jacktasia/dumb-jump#401]
# Prerequisites
When searching, `dumb-jump` first searches with context, and then tries without context if the context-search yielded no results.

Example: .php file, "<?php new Contact()" , cursor on "Contact"

1. Context is "class", first search is for "class Contact"
2. If no result is found, only "Contact" is searched.

The second search will yield much more results than the context-sensitive first search, because the search terms are much more unspecific.

# Problem
When the `.dumpjump` file contains an include for an external path like `+../external/`, then too many results are shown because the external directory is searched context-less, even if the project path has context-sensitive results.

# Example
- .dumbjump contains one line `+../external/`
- emacs has `.php` file open, "<?php new Contact()" , cursor on "Contact"

1. Context is class, first search in **project directory** is for "class Contact". A result is found.
2. Context-free search in **project directory** is skipped because 1. had results
2. Context is class, first search in **../external/ directory** is for "class Contact. No results found
3. Context-less search in  **../external/ directory**  for "Contact". Dozends of results found
4. dumb-jump shows a result popup with dozens of entries

# Desired solution
It would be better if the search order would be different:

1. Context search in project directory
2. Context search in external directory
3. If something has been found, stop here and show results
4. Context-free search in project directory
5. Context-free search in external directory
6. Show results

**Comment by cweiske** [comment]
@jacktasia is there some way I can pay you to fix this?

**Comment by jacktasia** [comment]
> @jacktasia is there some way I can pay you to fix this?

Nope, but PRs are welcome. I believe there are services out there that allow you to add bounties for requests on open source projects. There may be someone out that will do it if you can/will not either.
